machineconfig/master/spec/config/storage/files/haproxy.sh L3Zhci91c3Jsb2NhbC9iaW4vaGFwcm94eS5zaAo=

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+set -e
+
+
+function generate_haproxy_cfg {
+envsubst < $HOST_CFG_PATH/haproxy.cfg.template | sudo tee $HOST_CFG_PATH/haproxy.cfg
+
+etcd_members=$(dig +noall +answer -t SRV _etcd-server-ssl._tcp.$CLUSTER_DOMAIN | awk '{print $NF}')
+for item in $etcd_members
+do
+    ips=$(dig +noall +answer $item  | awk '{print $NF}')
+    if [[ "$?" -eq 0 && "${#ips[@]}" -ne 0 ]]; then
+        sudo echo "   server "$item" "$ips":"$OCP_API_PORT" weight 1 check inter 3s fall 3 rise 3" >>  $HOST_CFG_PATH/haproxy.cfg
+    fi
+done
+
+}
+
+function does_lb_topology_changed {
+etcd_members=$(dig +noall +answer -t SRV _etcd-server-ssl._tcp.$CLUSTER_DOMAIN   | awk '{print $NF}')
+for item in $etcd_members
+do
+    ips=$(dig +noall +answer $item | awk '{print $NF}')
+    haproxy_cfg_ip=($(grep $item $HOST_CFG_PATH/haproxy.cfg | awk '{print $3}' | sed 's/:.*//'))
+    if [[ "$?" -eq 0 && "${#ips[@]}" -ne 0 ]]; then
+       if [ "$haproxy_cfg_ip" != "$ips" ]; then
```

Comment:
shouldn't you be sorting before comparing? Otherwise if the answer gives the same answer but in a different order, it will give a false positive.

Reply:
@celebdor 
I trigger reconfigure haproxy only if :: 
1. etcd-x IP (resolved by DNS) != etcd-x ip in haproxy cfg
Or
2. Can't resolve etcd-x IP in DNS but etcd-x IP appears in haproxy-cfg
I think that it should be fine, am I missing something?